storage/filesystem.go

```diff
@@ -176,17 +189,33 @@ func (f *Filesystem) RegisterMetadata(meta metadata.Metadata) (bool, error) {
 	return false, nil
 }
 
-func (f *Filesystem) WriteFiles(volumeID string, files map[string][]byte) error {
-	if err := os.MkdirAll(f.dataPathForVolumeID(volumeID), 0644); err != nil {
+// WriteFiles writes the given data to filesystem files within the volume's
+// data directory. Filesystem supports changing ownership of the data directory
+// to a custom gid.
+func (f *Filesystem) WriteFiles(meta metadata.Metadata, files map[string][]byte) error {
+	// Data directory should be read, write and execute only to the fs user; read and executable to group
+	if err := os.MkdirAll(f.dataPathForVolumeID(meta.VolumeID), 0750); err != nil {
```

Comment:
Hm, _do_ we want to permit writing by the owning user? Does this trip things up for the CSI driver if not? How come `0644` worked before, and now it needs to be `0750?`

Reply:
Presumably within the container this wouldn't be writable by anyone anyway due to it being a read-only FS. But do we need to keep it this way to allow the driver to write the data?

Reply:
Changed this to `0550` which does the job.

We need it to be readable and executable by both the user and group. 

The reason it worked before is that it wasn't run against a container with a different user/group.